File: 0001-two-sum/0001-two-sum.go
func twoSum(nums []int, target int) []int {
    n := len(nums)
    
    cnt := map[int]int{}
    for i := 0 ; i < n; i ++{
        t := target - nums[i]
        if j, ok := cnt[t]; ok{
            return []int{j, i}
        }
        cnt[nums[i]] = i
    }

    return nil

}

File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:

        cnt = Counter()
        for i in range(len(nums)):
            t = target - nums[i]
            if t in cnt:
                return [cnt[t], i]
            
            cnt[nums[i]] = i


        